src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (DataProcessingException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot add actor " + actor);
+        } finally {
+            if (transaction != null) {
+                session.close();
+            }
+        }
+        return actor;
+
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Actor actor = session.get(Actor.class, id);
+            return Optional.ofNullable(actor);
+        } catch (DataProcessingException e) {
+            throw new RuntimeException("cannot find actor with " + id);
```

Comment:
same in other places
```suggestion
        } catch (Exception e) {
            throw new DataProcessingException("cannot find actor with " + id, e);
```